src/main/java/org/jenkinsci/plugins/workflow/actions/ExecutorTaskInfoAction.java

```diff
@@ -0,0 +1,103 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public class ExecutorTaskInfoAction extends InvisibleAction implements FlowNodeAction, PersistentAction {
+    private static final long serialVersionUID = 1;
+
+    private String whyBlocked;
```

Comment:
I defer to @svanoort - I was wary of adding too much to the action for fear of performance issues.